frame/staking/src/pallet/impls.rs

```diff
@@ -359,6 +366,9 @@ impl<T: Config> Pallet<T> {
 			// Set ending era reward.
 			<ErasValidatorReward<T>>::insert(&active_era.index, validator_payout);
 			T::RewardRemainder::on_unbalanced(T::Currency::issue(rest));
+
+			// Clear offending validators.
+			<OffendingValidators<T>>::take();
```

Comment:
```suggestion
			<OffendingValidators<T>>::kill();
```

or 

```suggestion
			let _ = <OffendingValidators<T>>::take();
```

Reply:
should kill for sure if the value is not used. should be more efficient